langchain4j-core/src/main/java/dev/langchain4j/web/search/WebSearchOrganicResult.java

```diff
@@ -0,0 +1,221 @@
+        this.url = ensureNotNull(url,"url");
+        this.snippet = snippet;
+        this.content = content;
+        this.metadata = getOrDefault(metadata, new HashMap<>());
+    }
+
+    /**
+     * Returns the title of the web page.
+     *
+     * @return The title of the web page.
+     */
+    public String title() {
+        return title;
+    }
+
+    /**
+     * Returns the URL associated with the web page.
+     *
+     * @return The URL associated with the web page.
+     */
+    public URI url() {
+        return url;
+    }
+
+    /**
+     * Returns the snippet associated of the web page.
+     *
+     * @return The snippet associated of the web page.
+     */
+    public String snippet() {
+        return snippet;
+    }
+
+    /**
+     * Returns the content related from the scraped url.
+     *
+     * @return The content related from the scraped url.
+     */
+    public String content() {
+        return content;
+    }
+
+    /**
+     * Returns the result metadata associated with the search result.
+     *
+     * @return The result metadata associated with the search result.
+     */
+    public Map<String, String> metadata() {
+        return metadata;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        WebSearchOrganicResult that = (WebSearchOrganicResult) o;
+        return Objects.equals(title, that.title)
+                && Objects.equals(url, that.url)
+                && Objects.equals(snippet, that.snippet)
+                && Objects.equals(content, that.content)
+                && Objects.equals(metadata, that.metadata);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(title, url, snippet, content, metadata);
+    }
+
+    @Override
+    public String toString() {
+        return "WebSearchOrganicResult{" +
+                "title='" + title + '\'' +
+                ", url=" + url +
+                ", snippet='" + snippet + '\'' +
+                ", content='" + content + '\'' +
+                ", metadata=" + metadata +
+                '}';
+    }
+
+    /**
+     * Converts this WebSearchOrganicResult to a TextSegment.
+     *
+     * @return The TextSegment representation of this WebSearchOrganicResult.
+     */
+    public TextSegment toTextSegment() {
+        return TextSegment.from(content != null ? content : snippet,
+                copyToMetadata());
+    }
+
+    /**
+     * Converts this WebSearchOrganicResult to a Document.
+     *
+     * @return The Document representation of this WebSearchOrganicResult.
+     */
+    public Document toDocument() {
+        return Document.from(content != null ? content : snippet,
+                copyToMetadata());
+    }
+
+    private Metadata copyToMetadata() {
+        Metadata docMetadata = new Metadata();
+        docMetadata.add("title", title);
```

Comment:
I am wondering... should title be prepended to the content/snippet instead?

`TextSegment` text:
```
{{title}}
{{content or snippet}}
```

Reply:
that is a good question. So far I was considering `{{title}} and {{url}}` as a `TextSegment` or `Document` metadata. Only `{{content or snippet}}` set `TextSegment` or `Document` text.
However, `{{title}} and {{url}}` are mandatories in the constructor and bcz `TextSegment` and `Document` are used for rag purposes, make sense avoid that `text` is empty or null, so we can modify it like this:

`TextSegment.text` and `Document.text`

```
{{title}}   // Not nullable
{{content or snippet}} // Nullable
```

`TextSegment.metadata` and `Document.metadata`

```
{ "key": ''url" }, { "value": "myurl" } // Not nullable
{ "key": ''metadata1" }, { "value": "metadata1" } // Nullable
```

if you're agree, I can apply these changes asap :)

Reply:
Yeah, I guess it is better to include `title` in the text of `TextSegment`/`Document`, as it will provide an additional context about the `snippet`/`content` to the scoring and/or language model.

And the metadata will contain url and custom metadata. I guess it is what you've suggested above, right?

Reply:
yeah, that's how it is.